openedx/openedx-platform
# feat: [FC-0044] expose editor for advanced xblocks and move modal in CMS

## Description

In this PR a new view is added to CMS to render xblock modal views such as editor for advanced problem types and move xBlock modal. It's intended to use this view to render mentioned earlier interfaces in the course-authoring MFE. The course-authoring implementation can be found in this PR - https://github.com/openedx/frontend-app-course-authoring/pull/985.
This particular decision was made because the functionality to edit and move modals depends on the BackboneJS implementation, and can't be recreated in the React.js within the current solution.

According to discussions in https://github.com/openedx/edx-platform/pull/34161#issuecomment-2112522240 and https://github.com/openedx/frontend-app-course-authoring/pull/964#pullrequestreview-2033590615 the final decision for rendering xblock previews will also affect rendering of the edit modal for advanced xBlock. Therefore, it was decided to postpone the merge of the current implementation, and revise it after the decision for xBlocks previews is made.

## Supporting information

Issue: https://github.com/openedx/platform-roadmap/issues/321

## Testing instructions

- Start CMS service
- Copy xBlock locator for advanced problem
- Open to - {CMS_BASE}/xblock/{xblock_locator}/actions/edit
- See edit modal for the xBlock

- Start CMS service
- Copy xBlock locator for advanced problem
- Open to - {CMS_BASE}/xblock/{xblock_locator}/actions/move
- See move modal for the xBlock

## Deadline

"None"

## Other information

This PR is a replacement for the github.com/openedx/edx-platform/pull/34656
